src/ray/util/internal/stream_redirection_utils.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "ray/util/pipe_logger.h"
+#include "ray/util/scoped_dup2_wrapper.h"
+
+namespace ray::internal {
+
+struct RedirectionHandleWrapper {
```

Comment:
The naming in this file is pretty inconsistent, can you help me understand the reasoning/differences between:

- "RedirectionHandle" used here (is it always a "stream"?)
- "StreamRedirection" used in the util below
- "RedirectedStream" used in the test-only util

Also -- what's the difference between "RedirectedStream" here and "StreamRedirection"

Reply:
Discussed offline, in the latest commit I use ctor/dtor to simplify the logic a lot.